module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1325,6 +1328,96 @@ public function callIlsWithFailover($methodName, $params)
         );
     }
 
+    /**
+     * Get timed blocks for a method from driver configuration
+     *
+     * @param string $methodName Method to check
+     *
+     * @return array Array with keys 'start', 'end', 'recurring'
+     *               or empty array if no blocks are found
+     */
+    public function getMethodTimedBlocks(string $methodName): array
+    {
+        $functionConfig = $this->checkCapability('getConfig', ['TimedBlocks'])
+            ? $this->getDriver()->getConfig('TimedBlocks')
+            : [];
+        if (empty($functionConfig)) {
+            return [];
+        }
+        $methodBlocks = [];
+        foreach ($functionConfig as $key => $value) {
+            [$method, $setting] = explode(':', $key, 2);
+            $methodBlocks[$method][$setting] = $value;
+        }
+        if (!isset($methodBlocks[$methodName])) {
+            return [];
+        }
+        $methodBlock = $methodBlocks[$methodName];
+        $startDate = strtotime($methodBlock['startDate'] ?? '');
+        $endDate = $methodBlock['endDate'] ?? '';
+        $noEndHours = empty(explode(' ', $endDate, 2)[1]);
+        if ($endDate && $noEndHours) {
+            $endDate .= ' 23.59.59';
+        }
+        $endDate = strtotime($endDate);
+        if ($startDate || $endDate) {
+            return [
+                'start' => $startDate,
+                'end' => $endDate,
+                'recurring' => false,
+            ];
+        }
+        if (
+            ($startTime = strtotime($methodBlock['recurringStart'] ?? ''))
+            && ($endTime = strtotime($methodBlock['recurringEnd'] ?? ''))
+        ) {
+            return [
+                'start' => $startTime,
+                'end' => $endTime,
+                'recurring' => true,
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Check whether a method is currently blocked in TimedBlocks section of
+     * driver configuration
+     *
+     * @param string $methodName Method to check
+     *
+     * @return bool
+     */
+    public function isMethodBlocked(string $methodName): bool
```

Comment:
Same question: does this need to be public? (Admittedly, making it public would make it easier to write a unit test, which I think would be a great idea due to  all the variations and edge cases -- but if testing is the only reason, we could cheat with the ReflectionTrait, so it's not that big of a deal).

Reply:
If these are public, they could be used to get the blocked times to display that information to the user somewhere in the UI

Reply:
True. But there's a lot of processing required to figure that out due to the number of possible settings. I wonder if having a public `getMethodTimedBlockEndDateTime` method (or something like that) would provide the useful external information in an easier-to-consume fashion. Or, alternatively, maybe we need a view helper (or other utility class) that can transform all of the settings in useful ways, so consumers of the method don't have to reinvent complicated logic.